GoogleForCreators/web-stories-wp
# Animation: Adding Animation panel to Editor

## Summary

This PR lays down the foundation for editing animations within the editor using animation panels!  You can `Add`, `Update`, and `Remove` animation effects using the animation panels in the editor.

## Relevant Technical Choices

We currently store `animation` data in an array at the `page` level, but the existing editor panel code is setup to easily update properties within an element.  So, instead of adding in a bunch of extra code to specifically update animations as a separate thing, I decided to treat animations as a property of an element so animation data can be pushed up like any other element data.

## To-do

Coming soon in another PR:
- Tests (karma and other tests).
- Logic to handle animation panels when multiple elements are selected.
- Some sort of support for `Animation Parts` (so far I'm specifically just supporting `Animation Effects`, which are the animations we want users to specifically use).
- Update the panel styles to appear like the mocks but that's going to be part of the greater editor design update.

## User-facing changes

None.

## Testing Instructions

1.) Enable animations
2.) Go to a story (or create a new one).
3.) Select an element (only 1 element at a time) and the "Animation" panel should appear in the panel list.
4.) Select an effect from the drop down to apply the effect (with default values) to the selected element.
5.) Add more effects, remove some, have fun.
6.) Click the play button under the editor page area and see the animations come to life!

## Screenshot

![image](https://user-images.githubusercontent.com/40646372/90823822-55a09280-e2eb-11ea-9a58-9beacd46a5b2.png)
